scripts/targetcli

```diff
@@ -172,27 +172,41 @@ def get_arguments(shell):
 
     if len(sys.argv) > 1:
         command = " ".join(sys.argv[1:])
+        call_daemon(shell, command.encode())
+        sys.exit(0)
+
+    if interactive:
+        shell.con.display("targetcli shell version %s\n"
+                          "Entering targetcli interactive mode for daemonized approach.\n"
+                          "Type 'exit' to quit.\n"
+                          % targetcli_version)
     else:
-        inputs = []
         shell.con.display("targetcli shell version %s\n"
-                        "Entering targetcli batch mode for daemonized approach.\n"
-                        "Enter multiple commands separated by newline and "
-                        "type 'exit' to run them all in one go.\n"
-                        % targetcli_version)
-        while True:
-            shell.con.raw_write("/> ")
-            command = six.moves.input()
-            if command.lower() == "exit":
-                break
+                          "Entering targetcli batch mode for daemonized approach.\n"
+                          "Enter multiple commands separated by newline and "
+                          "type 'exit' to run them all in one go.\n"
+                          % targetcli_version)
+
+    inputs = []
+    while True:
+        real_exit=False
+        shell.con.raw_write("/> ")
+        command = six.moves.input()
+        if command.lower() == "exit":
+            real_exit=True
+        if not interactive:
             inputs.append(command)
-        command = '%'.join(inputs)  # delimit multiple commands with '%'
-
-    if not command:
-        sys.exit(1)
-
-    usage_version(command);
+            command = '%'.join(inputs)  # delimit multiple commands with '%'
```

Comment:
We can move the Line#199 just before Line#204, no need to join it here every time.